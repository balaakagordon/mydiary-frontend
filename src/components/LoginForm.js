import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { loginAction } from '../actions/loginActions';
import { toast } from 'react-toastify';
import history from '../history';
import FormInputField from './FormComponents/FormInputField';
import FormSubmitButton from './FormComponents/FormSubmitButton';


export class LoginForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      user: {
        email: '',
        password: ''
      },
      loading: false,
      message: null,
      status: '',
      errors: null
    };

    this.handleFormInput = this.handleFormInput.bind(this);
    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleClearForm = this.handleClearForm.bind(this);
    this.handleSuccess = this.handleSuccess.bind(this);
    this.handleErrors = this.handleErrors.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.loading) {
      console.log('loading...');
    }
    if (nextProps.status === 'success') {
      this.handleSuccess(nextProps.message);
    } else if (nextProps.status === 'error') {
      this.handleErrors(nextProps.errors);
    }
  }

  handleFormSubmit(event) {
    event.preventDefault();
    this.props.userSignup(this.state.user);
  }

  handleClearForm = (event) => {
    event.preventDefault();
    this.setState({
      user: {
        firstName: '',
        lastName: '',
        email: '',
        password: '',
        confirmedPassword: ''
      },
      loading: false,
      message: null,
      status: '',
      token: '',
      errors: null
    })
  }

  handleFormInput(event) {
    let {name, value} = event.target;
    this.setState( prevState => {
        return {
        user :{
          ...prevState.user, [name]: value
        }
      }
    })
  }

  handleSuccess = message => {
    toast.success(message, {
      position: toast.POSITION.TOP_CENTER,
      hideProgressBar: true
    });
    history.push('/home');
  }

  handleErrors = errors => {
    for (let err in errors) {
      toast.error(errors[err][0], {
        position: toast.POSITION.TOP_CENTER,
        hideProgressBar:true
      });
    }
  };

  render() {
    return (
      <div className="container auth-form">
        <FormInputField type={'text'}
            title={'Email'}
            name={'email'}
            value={this.state.user.email}
            placeholder={'Enter your email address'}
            handleChange={this.handleFormInput}
        />
        <FormInputField type={'password'}
            title={'Password'}
            name={'password'}
            value={this.state.user.password}
            placeholder={'Enter your password'}
            handleChange={this.handleFormInput}
        />
        <Link to="/">don't have an account?</Link>
        <br />
        <FormSubmitButton
            action={this.handleFormSubmit}
            title={'Submit'}
        />
        <FormSubmitButton
            action={this.handleClearForm}
            title={'Cancel'}
        />
      </div>
    );
  }
}

LoginForm.propTypes = {
  loginAction: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  user: state.login.user,
  message: state.login.message,
  status: state.login.status
});

export default withRouter(
  connect(
    mapStateToProps,
    { loginAction }
  )(LoginForm)
);
