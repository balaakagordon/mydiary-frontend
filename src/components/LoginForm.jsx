import React from "react";
import { Link } from "react-router-dom";
import FormInputField from "./FormComponents/FormInputField";
import Button from "./FormComponents/Button";

const LoginForm = (props) => {
  return (
    <div className="auth-form">
      {props.showLoader}
      <FormInputField
        type={"text"}
        title={"Email"}
        name={"email"}
        value={props.user.email}
        placeholder={"Enter your email address"}
        handleChange={props.handleFormInput}
        className={'auth-form-input'}
      />
      <FormInputField
        type={"password"}
        title={"Password"}
        name={"password"}
        value={props.user.password}
        placeholder={"Enter your password"}
        handleChange={props.handleFormInput}
        className={'auth-form-input'}
      />
      <div className="auth-link">
        <Link to="/register">don't have an account?</Link>
      </div>
      <div className="row">
        <div className="form-buttons">
          <Button
            action={props.handleFormSubmit}
            title={"Submit"}
            className='auth-button auth-submit'
          />
          <Button
            action={props.handleClearForm}
            title={"Cancel"}
            className='auth-button auth-cancel'
          />

        </div>
      </div>
    </div>
  );
}

export default LoginForm;
