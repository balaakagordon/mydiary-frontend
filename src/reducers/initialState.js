export default {
    signup: {
        loading: false,
        message: null,
        status: '',
        token: '',
        errors: null
    },
    login: {
        loading: false,
        message: null,
        status: '',
        token: '',
        errors: null
    },
    getEntries: {
        entriesData: {
            entries: [],
            total: 0,
            currentPage: 1,
            from: null,
            to: null,
            lastPage: null,
            perPage: null,
            firstPageUrl: null,
            lastPageUrl: null,
            prevPageUrl: null,
            nextPageUrl: null
        },
        loading: false,
        message: null,
        status: null,
        errors: null
    },
    newEntry: {
        entry: {
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    updateEntry: {
        entry: {
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    getEntry: {
        entry: {
            id: null,
            author: null,
            title: '',
            body: '',
            lastEdited: null
        },
        loading: false,
        status: null,
        message: null,
        errors: null
    },
    getProfile: {
        profile: {
            id: null,
            firstName: null,
            lastName: null,
            email: null,
            image: null,
            lastUsed: null,
            currentEntries: null,
            allEntries: null,
            notifications: null,
            createdAt: null,
            // selectedFile: null
        },
        loading: false,
        status: null,
        errors: null
    },
    navbar: {
        authenticated: false,
    },
  };