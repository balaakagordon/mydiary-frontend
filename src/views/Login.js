import React from 'react';
import LoginForm from '../components/LoginForm'
import Header from '../components/Header'


const Login = () => (
	<div>
		<Header />
		<LoginForm />
	</div>
);

export default Login;