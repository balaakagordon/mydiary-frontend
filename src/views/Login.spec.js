import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import initialState from "../reducers/initialState";
import LoginForm from "../components/LoginForm";
import history from "../history";

import { Login, mapStateToProps } from "./Login";

describe("The Login Component", () => {
  describe("The container element", () => {
    it("should match the snapshot", () => {
      const tree = mapStateToProps(initialState);
      expect(tree).toMatchSnapshot();
    });

    it("should map the state to props correctly", () => {
      const sampleProps = {
        user: {
          email: "e.m@email.com",
          password: "Password"
        },
        loading: false,
        message: null,
        status: "",
        errors: null
      };

      const appState = {
        login: {
          ...sampleProps
        }
      };

      const ownProps = {
        user: {}
      };

      const componentState = mapStateToProps(appState, ownProps);
      expect(componentState).toEqual(sampleProps);
    });
  });

  const mockStore = configureStore();
  const store = mockStore(initialState.login);
  let component;
  beforeEach(() => {
    component = shallow(
      <Login store={store} {...initialState.login} />
    );
  });
  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    expect(component).toBeDefined();
  });

  it("should render children", () => {
    expect(component.find(LoginForm)).toBeDefined();
  });

  it("should call componentWillReceiveProps", () => {
    expect(component.find(LoginForm)).toBeDefined();
  });
});

describe("componentWillReceiveProps()", () => {
  it("is called after receiving props", () => {
    const nextProps = {
        status: "success"
    };
    let wrapper = shallow(<Login />);
    wrapper.instance().handleSuccess = jest.fn();
    wrapper.update();
    wrapper.setProps(nextProps);
    expect(wrapper.instance().handleSuccess).toBeCalled();
  });
});
