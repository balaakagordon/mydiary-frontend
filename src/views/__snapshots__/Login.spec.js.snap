// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Login Component The container element should match the snapshot 1`] = `
Object {
  "errors": null,
  "loading": false,
  "message": null,
  "status": "",
  "token": "",
  "user": undefined,
}
`;

exports[`The Login Component should render correctly 1`] = `
<div>
  <LoginForm
    handleClearForm={[Function]}
    handleErrors={[Function]}
    handleFormInput={[Function]}
    handleFormSubmit={[Function]}
    handleSuccess={[Function]}
    showLoader={null}
    user={
      Object {
        "email": "",
        "password": "",
      }
    }
  />
</div>
`;
